libtenzir/builtins/functions/community_id.cpp

```diff
@@ -0,0 +1,141 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2024 The Tenzir Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <tenzir/as_bytes.hpp>
+#include <tenzir/community_id.hpp>
+#include <tenzir/concept/parseable/tenzir/si.hpp>
+#include <tenzir/detail/narrow.hpp>
+#include <tenzir/flow.hpp>
+#include <tenzir/tql2/arrow_utils.hpp>
+#include <tenzir/tql2/plugin.hpp>
+
+#include <caf/fused_downstream_manager.hpp>
+
+namespace tenzir::plugins::community_id {
+
+namespace {
+
+struct arguments {
+  ast::expression src_ip;
+  ast::expression dst_ip;
+  ast::expression dst_port;
+  ast::expression src_port;
+  ast::expression proto;
+};
+
+class plugin final : public function_plugin {
+public:
+  auto name() const -> std::string override {
+    return "tql2.community_id";
+  }
+
+  auto make_function(invocation inv, session ctx) const
+    -> failure_or<function_ptr> override {
+    auto args = arguments{};
+    TRY(argument_parser2::function("community_id")
+          .add(args.src_ip, "<source ip>")
+          .add(args.dst_ip, "<destination ip>")
+          .add(args.src_port, "<source port>")
+          .add(args.dst_port, "<destination port>")
+          .add(args.proto, "<transport protocol>")
```

Comment:
The arguments in this are all positional, when they could also all be named. What do we want to default to here?

I personally think that named arguments are better.

Reply:
Agreed, in this case there is no obvious argument order. I would argue only then it's a better UX to have positional arguments.

Reply:
The given order here seems pretty established across the spec and various implementations. So I think it might be fine? It's also hard to misuse without at least getting a warning.

Reply:
Even though the spec has a fixed order, the reality is that networking monitoring tools present flows in different ways. I'd say we should switch to named arguments.

Reply:
All arguments are now named arguments.